tests/AutoReview/FixerFactoryTest.php

```diff
@@ -46,584 +46,191 @@ public function testFixersPriorityEdgeFixers(): void
         }
```

Comment:
no clue what changes are happening here :( how to review it? can you describe the changes, so one can easier understand what we look at?

Reply:
Sure, I opened it early to get some feedback and to see the CI and such.

I started this as a part of changing the fixers get-priority method from returning a seeming arbitrary integer value to returning fixer names of which it must be executed before (but that is a long way out still).

I started naming the code following the naming more typical in topological sorting.
Than to start prototyping this sorting (not part of the PR) I wanted to have the priority graph in an array in the format you can see here:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/6231/files#diff-a809d26a7ee19fceeb074078036812dc5c022e0a52d7ef0cc27d13cd161b33dfR299
I generated that graph from the current priority listing. From there I redid all the existing tests on top of that graph.

So the value this PR bring, all IMHO ;)
- new composition of the test is more easy to follow, exceptions stand more out
- adding a new fixer and its priorities in the future will require less changes to the test
- it will be more simple to start moving some of the exceptions cases listed in the test into the one new graph
- much later it might be more easy to actual change the fixer `getPriority` method based on the test
- the test uses far less memory and is faster (on my setup)

I tried to reduce the diff, but there is not much room. I think the easiest for review is to check the new file without literal checking against the current one.

Reply:
Thanks for the description! I guess it will, unfortunately, take quite a while to properly review this.


> I started this as a part of changing the fixers get-priority method from returning a seeming arbitrary integer value to returning fixer names of which it must be executed before (but that is a long way out still).

pls remember that if moving from hardcoded numbers to relations, we need to ensure custom fixers could plug-in to that mechanism too

Reply:
```
pls remember that if moving from hardcoded numbers to relations, we need to ensure custom fixers could plug-in to that mechanism too
```

yeah that is the `but that is a long way out still` part ;) 
but than again the only way to get there is by keep making small steps like this one ^_^


```
take quite a while to properly review this
```

sure thing, in the end it is the same test, just expressed in a different form :+1:  good luck!

Reply:
let me know if you find anything, than I'll patch :+1: